drivers/include/periph/gpio_ll_irq.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2020 Gunar Schorcht
+ *               2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_gpio_ll_irq  IRQ Support in Next Generation
+ *                                          Peripheral GPIO Interface
+ * @ingroup     drivers_periph_gpio_ll
+ * @brief       IRQ Support in Next Generation Peripheral GPIO Interface
+ *
+ * @warning     This API is not stable yet and intended for internal use only
+ *              as of now.
+ * @note        This API can only be used, if the features `periph_gpio_ll_irq`
+ *              is used. Not every MCU will implement the features required for
+ *              this API or will have a suitable driver
+ *              available.
+ *
+ * # Motivation
+ *
+ * The IRQ related GPIO-NG API is very similar to the one of @ref
+ * drivers_periph_gpio - however two aspects have been changed (in addition to
+ * using separate port and pin parameters): First, level triggered IRQs that
+ * many MCUs support are now exposed. (Note: They can also be trivially be
+ * emulated, see the STM32 implementation as example.) Second, the
+ * configuration of the GPIO pin and the associated IRQ have been split into
+ * two functions. This allows reconfiguring the trigger of an IRQ with less
+ * overhead. This can result in reconfiguring the IRQ trigger being less effort
+ * than filtering out unneeded events and hence consume power due to longer
+ * power saving phases.
+ *
+ * @{
+ * @file
+ * @brief       IRQ Support in Next Generation Peripheral GPIO Interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @warning     This API is not stable yet and intended for internal use only
+ *              as of now.
+ */
+
+#ifndef PERIPH_GPIO_LL_IRQ_H
+#define PERIPH_GPIO_LL_IRQ_H
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "periph/gpio_ll.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(HAS_GPIO_LL_IRQ_UNMASK) || defined(DOXYGEN)
+/**
+ * @brief   Indicates whether (1) or not (0) IRQs can be detected while masked
+ *
+ * If this resolves to `1`, @ref gpio_ll_irq_unmask is provided. Otherwise, only
+ * @ref gpio_ll_irq_unmask_clear is available.
+ */
+#define HAS_GPIO_LL_IRQ_UNMASK 1
```

Comment:
Isn't this a configuration that would have to be modelled as feature in ´Kconfig`?

Reply:
That would allow a driver to depend on this feature, which would be nice. Yes, I think that is better!